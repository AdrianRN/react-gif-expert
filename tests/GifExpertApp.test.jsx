import { fireEvent, render, screen } from "@testing-library/react";
import GifExpertApp from "../src/GifExpertApp";

describe('Tests in <GifExpertApp />', () => { 
    
    test('Check if text changes when writing in input', () => {
        
        render( <GifExpertApp /> );
        const input = screen.getByRole('textbox');

        fireEvent.input(input, { target: { value: 'Pikachu' } });

        expect(input.value).toBe('Pikachu');
        // screen.debug();
    });

    test('should match with snapshot', () => {
        
        const {container} = render(<GifExpertApp />);
        expect(container).toMatchSnapshot();
    });

    test('should add a new category', () => {
        
        render(<GifExpertApp />);
        const input = screen.getByRole('textbox');
        const form = screen.getByRole('form');

        fireEvent.input(input, { target: { value: 'Pikachu' } });
        fireEvent.submit(form);

        expect(screen.getByText('Pikachu'));
        // screen.debug();
    });

    test('should not add repeated category ', () => {
        
        render(<GifExpertApp />);
        const input = screen.getByRole('textbox');
        const form = screen.getByRole('form');

        // Eevee already exists
        fireEvent.input(input, { target: { value: 'eevee' } });
        fireEvent.submit(form);

        const count = screen.getAllByText('eevee')
        expect(count.length).toEqual(1);

        screen.debug();
    });


 });