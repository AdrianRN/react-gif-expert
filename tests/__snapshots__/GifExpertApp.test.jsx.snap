// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <GifExpertApp /> should match with snapshot 1`] = `
<div>
  <h1>
    GifExpertApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Search Gifs"
      type="text"
      value=""
    />
  </form>
  <h3>
    eevee
  </h3>
  <h2>
    Cargando ...
  </h2>
  <div
    class="card-grid"
  />
</div>
`;
